### ISSUE TITLE
MySQL port is hardcoded

### ISSUE THREAD

**Issue by maordavidov** [AzBuilder/terrakube#481]
### Bug description 🐞

MySQL port is hardcoded. Some managed services are not allowing to use the default port and we can't change it from the configuration, so we can't connect to the DB.

```
  case MYSQL:
                DriverManagerDataSource mysql = new DriverManagerDataSource();
                mysql.setDriverClassName("com.mysql.cj.jdbc.Driver");
                mysql.setUrl(String.format("jdbc:mysql://%s:3306/%s?serverTimezone=UTC", dataSourceConfigurationProperties.getHostname(), dataSourceConfigurationProperties.getDatabaseName()));
                mysql.setUsername(dataSourceConfigurationProperties.getDatabaseUser());
                mysql.setPassword(dataSourceConfigurationProperties.getDatabasePassword());
                dataSource = mysql;
                break;
```

```
https://github.com/AzBuilder/terrakube/blob/7f63e2af4535dd2c79e9a764f198c8ce1a8741c7/api/src/main/java/org/terrakube/api/plugin/datasource/DataSourceAutoConfiguration.java#L50C61-L50C65
```

### Steps to reproduce

Try to connect to any MySQL with non default port number

**Comment by alfespa17** [comment]
This look simple but it will require some changes.

Adding the port property here
https://github.com/AzBuilder/terrakube/blob/main/api/src/main/java/org/terrakube/api/plugin/datasource/DataSourceConfigurationProperties.java

Using the new property in dataSourceConfigurationProperties instead of the hardcoded value
https://github.com/AzBuilder/terrakube/blob/main/api/src/main/java/org/terrakube/api/plugin/datasource/DataSourceAutoConfiguration.java#L50

Adding a new property for the database port in the application properties using some new environment variable for the port like this
https://github.com/AzBuilder/terrakube/blob/main/api/src/main/resources/application.properties#L47

And finally update the helm chart with the new option to set the port here
https://github.com/AzBuilder/terrakube-helm-chart/blob/main/charts/terrakube/templates/secrets-api.yaml#L87

Would you like to help with the change and test if with those changes you are able to connect to your managed services?

**Comment by alfespa17** [comment]
Sure @stefano-maglione, all help is welcome you could fork the repository and create a pull request to fix the issue 👍

**Comment by alfespa17** [comment]
I can also assign the issue to you if you want

**Comment by stefano-maglione** [comment]
@alfespa17 yes please

**Comment by stefano-maglione** [comment]
@alfespa17 is there any wiki about how to run the project locally?

**Comment by alfespa17** [comment]
We have  this document so you can run the app in GITPOD using your browser, so you dont have to worry about the local environment setup and you can test the app really quick with just 1 click.

GITPOD gives you 50 hours for free, and you can use several tools like docker inside to test

https://github.com/AzBuilder/terrakube/blob/main/development.md

There is one script that you can run like this: 

```bash
./scripts/setupDevelopmentEnvironment.sh
```

It will create files with the environment variables that you need to start the components (.envApi , .envExecutor, .envRegistry, .envUi)

You will have to do some small change in the .envApi to change the environment variables to connect to a MYSQL

I hope this can help you.

**Comment by alfespa17** [comment]
Hello @maordavidov 

This issue was fixed in Terrakube 2.15.3 and helm chart 3.8.3

Thanks to @stefano-maglione for the help.
